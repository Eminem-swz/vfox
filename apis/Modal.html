<!DOCTYPE html>
<html lang="en-US">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <title>Modal 模态框弹窗 | Vfox for Vue3</title>
    <meta name="generator" content="VuePress 1.7.1">
    
    <meta name="description" content="移动端 Vue 组件库">
    
    <link rel="preload" href="/vfox/assets/css/0.styles.07e4087f.css" as="style"><link rel="preload" href="/vfox/assets/js/app.11c5dd7b.js" as="script"><link rel="preload" href="/vfox/assets/js/2.f96c0e2a.js" as="script"><link rel="preload" href="/vfox/assets/js/12.b4800caa.js" as="script"><link rel="prefetch" href="/vfox/assets/js/10.4d3031aa.js"><link rel="prefetch" href="/vfox/assets/js/11.b8e9d0e5.js"><link rel="prefetch" href="/vfox/assets/js/13.835e5e58.js"><link rel="prefetch" href="/vfox/assets/js/14.67692712.js"><link rel="prefetch" href="/vfox/assets/js/15.52f7507b.js"><link rel="prefetch" href="/vfox/assets/js/16.3706259a.js"><link rel="prefetch" href="/vfox/assets/js/17.71052dd5.js"><link rel="prefetch" href="/vfox/assets/js/18.cad7cf19.js"><link rel="prefetch" href="/vfox/assets/js/19.6ffb4022.js"><link rel="prefetch" href="/vfox/assets/js/20.cbba3ac5.js"><link rel="prefetch" href="/vfox/assets/js/21.456f3b58.js"><link rel="prefetch" href="/vfox/assets/js/22.8ef4a519.js"><link rel="prefetch" href="/vfox/assets/js/23.16665c41.js"><link rel="prefetch" href="/vfox/assets/js/24.bc7a3e0d.js"><link rel="prefetch" href="/vfox/assets/js/25.8200d57a.js"><link rel="prefetch" href="/vfox/assets/js/26.0a2e60cc.js"><link rel="prefetch" href="/vfox/assets/js/27.021711b6.js"><link rel="prefetch" href="/vfox/assets/js/28.12ef3780.js"><link rel="prefetch" href="/vfox/assets/js/29.37ce4854.js"><link rel="prefetch" href="/vfox/assets/js/3.ab10530e.js"><link rel="prefetch" href="/vfox/assets/js/30.4bf82915.js"><link rel="prefetch" href="/vfox/assets/js/31.843efbcb.js"><link rel="prefetch" href="/vfox/assets/js/32.d4803146.js"><link rel="prefetch" href="/vfox/assets/js/33.f969585f.js"><link rel="prefetch" href="/vfox/assets/js/34.d34301f8.js"><link rel="prefetch" href="/vfox/assets/js/35.8a3d4475.js"><link rel="prefetch" href="/vfox/assets/js/36.0111daed.js"><link rel="prefetch" href="/vfox/assets/js/37.cb0343e7.js"><link rel="prefetch" href="/vfox/assets/js/38.a3d7856b.js"><link rel="prefetch" href="/vfox/assets/js/39.fa65cd15.js"><link rel="prefetch" href="/vfox/assets/js/4.c2adc090.js"><link rel="prefetch" href="/vfox/assets/js/40.10054c93.js"><link rel="prefetch" href="/vfox/assets/js/41.0fd1a91c.js"><link rel="prefetch" href="/vfox/assets/js/42.ba2bc92c.js"><link rel="prefetch" href="/vfox/assets/js/43.f7e67e75.js"><link rel="prefetch" href="/vfox/assets/js/44.cd76ef03.js"><link rel="prefetch" href="/vfox/assets/js/45.bd4798dd.js"><link rel="prefetch" href="/vfox/assets/js/46.64e67ecc.js"><link rel="prefetch" href="/vfox/assets/js/47.ad3c0645.js"><link rel="prefetch" href="/vfox/assets/js/48.a63c007e.js"><link rel="prefetch" href="/vfox/assets/js/49.86e359dc.js"><link rel="prefetch" href="/vfox/assets/js/5.8ae9a8f0.js"><link rel="prefetch" href="/vfox/assets/js/50.8a817d9e.js"><link rel="prefetch" href="/vfox/assets/js/51.6f439c92.js"><link rel="prefetch" href="/vfox/assets/js/52.880c7f21.js"><link rel="prefetch" href="/vfox/assets/js/53.d8ddb489.js"><link rel="prefetch" href="/vfox/assets/js/54.85e6a146.js"><link rel="prefetch" href="/vfox/assets/js/55.c85faa47.js"><link rel="prefetch" href="/vfox/assets/js/56.d684c19b.js"><link rel="prefetch" href="/vfox/assets/js/57.b6ec83e4.js"><link rel="prefetch" href="/vfox/assets/js/58.60d3a792.js"><link rel="prefetch" href="/vfox/assets/js/59.2cba5254.js"><link rel="prefetch" href="/vfox/assets/js/6.5ddd5f2e.js"><link rel="prefetch" href="/vfox/assets/js/60.42f50da6.js"><link rel="prefetch" href="/vfox/assets/js/61.c25a6034.js"><link rel="prefetch" href="/vfox/assets/js/62.e20b0dcf.js"><link rel="prefetch" href="/vfox/assets/js/63.5b1087af.js"><link rel="prefetch" href="/vfox/assets/js/64.ec0cb62a.js"><link rel="prefetch" href="/vfox/assets/js/65.a5ca0ea6.js"><link rel="prefetch" href="/vfox/assets/js/66.6674684a.js"><link rel="prefetch" href="/vfox/assets/js/67.c9cfed38.js"><link rel="prefetch" href="/vfox/assets/js/68.85776dac.js"><link rel="prefetch" href="/vfox/assets/js/69.28fb2927.js"><link rel="prefetch" href="/vfox/assets/js/7.03a52a57.js"><link rel="prefetch" href="/vfox/assets/js/70.65843254.js"><link rel="prefetch" href="/vfox/assets/js/71.5ffb497f.js"><link rel="prefetch" href="/vfox/assets/js/72.2489fdda.js"><link rel="prefetch" href="/vfox/assets/js/73.37d54757.js"><link rel="prefetch" href="/vfox/assets/js/74.0c889b0d.js"><link rel="prefetch" href="/vfox/assets/js/75.30b27a41.js"><link rel="prefetch" href="/vfox/assets/js/76.d1a12202.js"><link rel="prefetch" href="/vfox/assets/js/77.c583b74c.js"><link rel="prefetch" href="/vfox/assets/js/78.48dd1c4e.js"><link rel="prefetch" href="/vfox/assets/js/79.41b5403a.js"><link rel="prefetch" href="/vfox/assets/js/8.47327716.js"><link rel="prefetch" href="/vfox/assets/js/80.905fcf4c.js"><link rel="prefetch" href="/vfox/assets/js/81.084f480c.js"><link rel="prefetch" href="/vfox/assets/js/82.6b5c642b.js"><link rel="prefetch" href="/vfox/assets/js/83.21f05420.js"><link rel="prefetch" href="/vfox/assets/js/84.4278840d.js"><link rel="prefetch" href="/vfox/assets/js/85.b4f255ae.js"><link rel="prefetch" href="/vfox/assets/js/86.0558f824.js"><link rel="prefetch" href="/vfox/assets/js/87.5369a89a.js"><link rel="prefetch" href="/vfox/assets/js/88.b7a26cef.js"><link rel="prefetch" href="/vfox/assets/js/89.dd29dc01.js"><link rel="prefetch" href="/vfox/assets/js/9.8d376d76.js"><link rel="prefetch" href="/vfox/assets/js/90.64005830.js"><link rel="prefetch" href="/vfox/assets/js/91.5aad9188.js">
    <link rel="stylesheet" href="/vfox/assets/css/0.styles.07e4087f.css">
  </head>
  <body>
    <div id="app" data-server-rendered="true"><div class="theme-container"><header class="navbar"><div class="sidebar-button"><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" role="img" viewBox="0 0 448 512" class="icon"><path fill="currentColor" d="M436 124H12c-6.627 0-12-5.373-12-12V80c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12z"></path></svg></div> <a href="/vfox/" class="home-link router-link-active"><img src="/vfox/assets/images/logo.png" alt="Vfox for Vue3" class="logo"> <span class="site-name can-hide">Vfox for Vue3</span></a> <div class="links"><div class="search-box"><input aria-label="Search" autocomplete="off" spellcheck="false" value=""> <!----></div> <nav class="nav-links can-hide"><div class="nav-item"><a href="/vfox/" class="nav-link">
  指南
</a></div><div class="nav-item"><a href="/vfox/design/" class="nav-link">
  设计
</a></div><div class="nav-item"><a href="/vfox/components/Button.html" class="nav-link">
  组件
</a></div><div class="nav-item"><a href="/vfox/apis/LocalStorage.html" class="nav-link">
  API
</a></div><div class="nav-item"><div class="dropdown-wrapper"><button type="button" aria-label="选择版本" class="dropdown-title"><span class="title">选择版本</span> <span class="arrow down"></span></button> <button type="button" aria-label="选择版本" class="mobile-dropdown-title"><span class="title">选择版本</span> <span class="arrow right"></span></button> <ul class="nav-dropdown" style="display:none;"><li class="dropdown-item"><!----> <a href="https://cdn.fox2.cn/1.x/docs/" target="_blank" rel="noopener noreferrer" class="nav-link external">
  1.x
  <span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a></li><li class="dropdown-item"><!----> <a href="/vfox/" class="nav-link">
  2.x
</a></li></ul></div></div> <!----></nav></div></header> <div class="sidebar-mask"></div> <aside class="sidebar"><nav class="nav-links"><div class="nav-item"><a href="/vfox/" class="nav-link">
  指南
</a></div><div class="nav-item"><a href="/vfox/design/" class="nav-link">
  设计
</a></div><div class="nav-item"><a href="/vfox/components/Button.html" class="nav-link">
  组件
</a></div><div class="nav-item"><a href="/vfox/apis/LocalStorage.html" class="nav-link">
  API
</a></div><div class="nav-item"><div class="dropdown-wrapper"><button type="button" aria-label="选择版本" class="dropdown-title"><span class="title">选择版本</span> <span class="arrow down"></span></button> <button type="button" aria-label="选择版本" class="mobile-dropdown-title"><span class="title">选择版本</span> <span class="arrow right"></span></button> <ul class="nav-dropdown" style="display:none;"><li class="dropdown-item"><!----> <a href="https://cdn.fox2.cn/1.x/docs/" target="_blank" rel="noopener noreferrer" class="nav-link external">
  1.x
  <span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a></li><li class="dropdown-item"><!----> <a href="/vfox/" class="nav-link">
  2.x
</a></li></ul></div></div> <!----></nav>  <ul class="sidebar-links"><li><a href="/vfox/" aria-current="page" class="sidebar-link">指南</a></li><li><a href="/vfox/design/" class="sidebar-link">设计</a></li><li><a href="/vfox/components/Button.html" class="sidebar-link">组件</a></li><li><section class="sidebar-group depth-0"><p class="sidebar-heading open"><span>API</span> <!----></p> <ul class="sidebar-links sidebar-group-items"><li><a href="/vfox/apis/LocalStorage.html" class="sidebar-link">LocalStorage 本地缓存</a></li><li><a href="/vfox/apis/Notify.html" class="sidebar-link">Notify 通知类型弹窗</a></li><li><a href="/vfox/apis/Toast.html" class="sidebar-link">Toast 消息类型弹窗</a></li><li><a href="/vfox/apis/Scroll.html" class="sidebar-link">Scroll 滚动</a></li><li><a href="/vfox/apis/Preview.html" class="sidebar-link">Preview 预览类型弹窗</a></li><li><a href="/vfox/apis/Modal.html" aria-current="page" class="active sidebar-link">Modal 模态框弹窗</a><ul class="sidebar-sub-headers"><li class="sidebar-sub-header"><a href="/vfox/apis/Modal.html#dialog-showdialog-object" class="sidebar-link">Dialog.showDialog(object)</a><ul class="sidebar-sub-headers"><li class="sidebar-sub-header"><a href="/vfox/apis/Modal.html#params" class="sidebar-link">Params</a></li><li class="sidebar-sub-header"><a href="/vfox/apis/Modal.html#usage" class="sidebar-link">Usage</a></li></ul></li></ul></li><li><a href="/vfox/apis/Drawer.html" class="sidebar-link">Drawer 抽屉类型弹窗</a></li><li><a href="/vfox/apis/Popover.html" class="sidebar-link">Popover 气泡类型弹窗</a></li></ul></section></li></ul> </aside> <main class="page"> <div class="theme-default-content content__default"><h1 id="modal-模态框弹窗"><a href="#modal-模态框弹窗" class="header-anchor">#</a> Modal 模态框弹窗</h1> <h2 id="dialog-showdialog-object"><a href="#dialog-showdialog-object" class="header-anchor">#</a> Dialog.showDialog(object)</h2> <p>显示模态对话框。</p> <h3 id="params"><a href="#params" class="header-anchor">#</a> Params</h3> <p>Object object</p> <table><thead><tr><th>属性</th> <th>类型</th> <th>默认值</th> <th>必填</th> <th>说明</th></tr></thead> <tbody><tr><td>title</td> <td>string</td> <td></td> <td>否</td> <td>提示的标题</td></tr> <tr><td>content</td> <td>string</td> <td></td> <td>是</td> <td>提示的内容</td></tr> <tr><td>maskClosable</td> <td>boolean</td> <td>false</td> <td>否</td> <td>点击蒙层是否触发关闭操作</td></tr> <tr><td>showCancel</td> <td>boolean</td> <td>true</td> <td>否</td> <td>是否显示取消按钮</td></tr> <tr><td>confirmText</td> <td>string</td> <td>'确定'</td> <td>否</td> <td>确认按钮的文字</td></tr> <tr><td>cancelText</td> <td>string</td> <td>'取消'</td> <td>否</td> <td>取消按钮的文字</td></tr> <tr><td>success</td> <td>function</td> <td></td> <td>否</td> <td>接口调用成功（在用户做出选择后，如取消，确定）的回调函数</td></tr> <tr><td>fail</td> <td>function</td> <td></td> <td>否</td> <td>接口调用失败（如传入错误的参数）的回调函数（不传入 fail 遇错误直接抛出）</td></tr> <tr><td>complete</td> <td>function</td> <td></td> <td>否</td> <td>弹窗关闭或调用失败的回调函数</td></tr></tbody></table> <h4 id="object-success-回调参数"><a href="#object-success-回调参数" class="header-anchor">#</a> object.success 回调参数</h4> <table><thead><tr><th>属性</th> <th>类型</th> <th>说明</th></tr></thead> <tbody><tr><td>confirm?</td> <td>boolean</td> <td>为 true 时，表示用户点击了确定按钮</td></tr> <tr><td>cancel?</td> <td>boolean</td> <td>为 true 时，表示用户取消了</td></tr></tbody></table> <h3 id="usage"><a href="#usage" class="header-anchor">#</a> Usage</h3> <div class="language- extra-class"><pre class="language-text"><code>import { Dialog } from 'vfox'

Dialog.showDialog({
  title: '提示',
  content: '这是一个模态弹窗',
  success (res) {
    if (res.confirm) {
      console.log('用户点击确定')
    } else if (res.cancel) {
      console.log('用户点击取消')
    }
  }
})
</code></pre></div></div> <footer class="page-edit"><!----> <!----></footer> <div class="page-nav"><p class="inner"><span class="prev">
      ←
      <a href="/vfox/apis/Preview.html" class="prev">
        Preview 预览类型弹窗
      </a></span> <span class="next"><a href="/vfox/apis/Drawer.html">
        Drawer 抽屉类型弹窗
      </a>
      →
    </span></p></div> </main></div><div class="global-ui"></div></div>
    <script src="/vfox/assets/js/app.11c5dd7b.js" defer></script><script src="/vfox/assets/js/2.f96c0e2a.js" defer></script><script src="/vfox/assets/js/12.b4800caa.js" defer></script>
  </body>
</html>
