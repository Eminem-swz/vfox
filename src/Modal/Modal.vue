<template>
  <teleport to="body">
    <div
      class="fx-modal fx-popup"
      :class="{ visible: visible2 }"
      :style="popupStyles"
      v-bind="$attrs"
      v-show="isShow"
    >
      <div class="fx-mask" @click="onMaskClick"></div>
      <div
        class="fx-modal_box"
        :style="{
          width
        }"
      >
        <div class="fx-modal_box-inner">
          <slot></slot>
        </div>
        <i v-if="showClose" class="fx-modal_close" @click="onCloseClick">
          <Icon icon="CloseCircleFilled" />
        </i>
      </div>
    </div>
  </teleport>
</template>

<script lang="ts">
import { defineComponent } from 'vue'
import { popupEmits, popupProps, usePopup } from '@/hooks/popup'
import Icon from '@/Icon'

export default defineComponent({
  name: 'fx-modal',
  components: { Icon },
  props: {
    ...popupProps,
    width: {
      type: String,
      default: null
    },
    showClose: {
      type: Boolean,
      default: true
    }
  },
  emits: popupEmits,
  setup(props, ctx) {
    const popup = usePopup(props, ctx, {})

    return {
      ...popup
    }
  }
})
</script>
