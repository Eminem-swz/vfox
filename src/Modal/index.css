.fx-modal_box {
  width: 400px;
  box-sizing: border-box;
  transform: scale(0);
  transition: all 0.2s;
  opacity: 0;
  position: relative;
}

.fx-modal_box-inner {
  min-height: 100px;
  border-radius: 4px;
  background-color: #fff;
  overflow: hidden;
}

.fx-modal.visible .fx-modal_box {
  opacity: 1;
  transform: scale(1);
}

.fx-modal_close {
  position: absolute;
  left: 50%;
  bottom: -50px;
  margin: 0 0 0 -16px;
  width: 32px;
  height: 32px;
  border-radius: 50%;
  display: flex;
  align-items: center;
  justify-content: center;
  box-sizing: border-box;
  cursor: pointer;
}

.fx-modal_close .fx-icon {
  --size: 32px;
  --color: #262626;
}

@media screen and (max-width: 575px) {
  .fx-modal_box {
    width: calc(100% - 96px);
  }
}
