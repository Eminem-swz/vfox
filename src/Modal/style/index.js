import '../../style/popup.scss'
import '../../Icon/index.scss'
import '../index.scss'
