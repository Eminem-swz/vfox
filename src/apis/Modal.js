import {
  isObject,
  isString,
  isUndefined,
  isFunction,
  noop
} from '../helpers/util'

function htmlEscape(text) {
  return text.replace(/[<>"&]/g, function(match) {
    switch (match) {
      case '<':
        return '&lt;'
      case '>':
        return '&gt;'
      case '&':
        return '&amp;'
      case '"':
        return '&quot;'
    }
  })
}

function render(options) {
  const html = [
    `<div class="modal">`,
    `<div class="modal-box">`,
    `<div class="modal-header">`,
    `<span class="modal-title">${options.title}</span>`,
    `</div>`,
    `<div class="modal-body">`,
    `<div class="modal-content">`,
    htmlEscape(options.content),
    `</div>`,
    `</div>`,
    `<div class="modal-footer">`,
    `<button style="display: ` +
      (options.showCancel ? 'block' : 'none') +
      `;" class="modal-button modal-cancel-button">${options.cancelText}</button>`,
    `<button class="modal-button modal-confirm-button primary">${options.confirmText}</button>`,
    `</div>`,
    `</div>`,
    `</div>`
  ].join('')

  const $wrapper = document.createElement('div')
  $wrapper.innerHTML = html

  return $wrapper.firstChild
}

var $el = null

const Events = {
  onCancelClick() {
    const res = {
      confirm: false,
      cancel: true
    }

    this._options.success(res)
    this._options.complete(res)

    hide()
  },
  onConfirmClick() {
    const res = {
      confirm: true,
      cancel: false
    }

    this._options.success(res)
    this._options.complete(res)

    hide()
  }
}

function getParamValue(object, key, type, defaultValue) {
  const param = object[key]

  if (!isUndefined(param)) {
    if (typeof param === type) {
      return param
    } else {
      throw new TypeError(
        `Invalid param: "object.${key}" must be a ${type} type.`
      )
    }
  } else {
    return defaultValue
  }
}

export function show(object) {
  if (!isObject(object)) {
    object = {}
  }

  const options = {}
  options.success = isFunction(object.success) ? object.success : noop
  options.fail = isFunction(object.fail) ? object.fail : noop
  options.complete = isFunction(object.complete) ? object.complete : noop

  try {
    if (!isString(object.content)) {
      throw new TypeError(
        'Invalid param: "object.content" must be a string type.'
      )
    }
    options.content = object.content
    options.title = getParamValue(object, 'title', 'string', '提示')
    options.maskClosable = getParamValue(
      object,
      'maskClosable',
      'boolean',
      false
    )
    options.showCancel = getParamValue(object, 'showCancel', 'boolean', true)
    options.confirmText = getParamValue(object, 'confirmText', 'string', '确定')
    options.cancelText = getParamValue(object, 'cancelText', 'string', '取消')

    hide()

    // 获取节点
    $el = render(options)

    // 添加事件
    const $cancel = $el.querySelector('.modal-cancel-button')
    $cancel._options = options
    $cancel.addEventListener('click', Events.onCancelClick, false)
    const $confirm = $el.querySelector('.modal-confirm-button')
    $confirm._options = options
    $confirm.addEventListener('click', Events.onConfirmClick, false)

    document.body.appendChild($el)
  } catch (e) {
    const res = {
      errMsg: e.message
    }

    if (options.fail != noop) {
      options.fail(res)
      options.complete(res)
    } else {
      options.complete(res)
      throw e
    }
  }
}

export function hide() {
  if ($el) {
    // 添加事件
    $el
      .querySelector('.modal-cancel-button')
      .removeEventListener('click', Events.onCancelClick, false)
    $el
      .querySelector('.modal-confirm-button')
      .removeEventListener('click', Events.onConfirmClick, false)

    document.body.removeChild($el)
    $el = null
  }
}
