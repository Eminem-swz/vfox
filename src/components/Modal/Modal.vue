<template>
  <div v-show="visibility" class="modal" @click="onModalClick">
    <div class="modal-box" @click.stop="onBoxClick">
      <div class="modal-header">
        <span class="modal-title">{{ title }}</span>
        <span class="modal-close" v-show="showClose" @click="onCloseClick">
          <icon type="close"></icon>
        </span>
      </div>
      <div class="modal-body">
        <div class="modal-content">
          <slot></slot>
        </div>
      </div>
      <div class="modal-footer">
        <button
          v-show="showCancel"
          class="modal-button modal-cancel-button"
          @click="onCancelClick"
        >
          {{ cancelText }}
        </button>
        <button
          class="modal-button modal-confirm-button primary"
          @click="onConfirmClick"
        >
          {{ confirmText }}
        </button>
      </div>
    </div>
  </div>
</template>

<script>
import Icon from '../Icon/Icon.vue'
import { getHandleEvent } from '../../helpers/events'
import './modal'

export default {
  name: 'app-modal',
  components: { Icon },
  props: {
    visibility: {
      type: Boolean,
      default: false
    },
    title: {
      type: String,
      default: '提示'
    },
    cancelText: {
      type: String,
      default: '取消'
    },
    confirmText: {
      type: String,
      default: '确定'
    },
    maskClosable: {
      type: Boolean,
      default: false
    },
    showClose: {
      type: Boolean,
      default: true
    },
    showCancel: {
      type: Boolean,
      default: true
    }
  },
  data() {
    return {}
  },
  created() {},
  beforeDestroy() {},
  methods: {
    onBoxClick() {},
    onModalClick(e) {
      if (this.maskClosable) {
        this.close(e)
      }
    },
    onCloseClick(e) {
      this.close(e)
    },
    onCancelClick(e) {
      const type = 'cancel'
      this.$emit(type, getHandleEvent(this.$el, e, {}, type))

      this.close(e)
    },
    onConfirmClick(e) {
      const type = 'confirm'
      this.$emit(type, getHandleEvent(this.$el, e, {}, type))

      this.close(e)
    },
    close(e) {
      this.$emit('update:visibility', false)

      const type = 'close'
      this.$emit(type, getHandleEvent(this.$el, e, {}, type))
    }
  }
}
</script>

<style>
@import url('../../global.css');

.modal {
  --confirm-background-color: var(--app-main-color);
  --confirm-border-color: var(--app-main-color);
  --confirm-font-color: #fff;
  --cancel-font-color: var(--app-semi-color);
  --cancel-background-color: #fff;
  --cancel-border-color: var(--app-light-color);

  position: fixed;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  z-index: 10000;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: rgba(0, 0, 0, 0.25);
  text-align: left;
  transform: translateZ(0);
}

.modal-box {
  width: 400px;
  box-sizing: border-box;
  border: 1px solid var(--app-light-color);
  border-radius: 4px;
  background-color: #fff;
  overflow: hidden;
}

.modal-header {
  display: flex;
  align-items: center;
  height: 51px;
  padding: 0 0 0 20px;
}

.modal-header .modal-title {
  flex: 1;
  font-size: 16px;
  color: var(--app-semi-color);
  margin-top: 5px;
  display: -webkit-box;
  -webkit-line-clamp: 1;
  -webkit-box-orient: vertical;
}

.modal-header .modal-close {
  width: 30px;
  height: 30px;
  margin-bottom: 21px;
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
}

.modal-header .modal-close .icon {
  width: 20px;
  height: 20px;
  fill: var(--app-light-color);
}

.modal-body {
  padding: 0 20px;
  overflow: hidden;
}

.modal-content {
  max-height: 300px;
  overflow: auto;
  font-size: 14px;
  line-height: 1.715em;
  color: var(--app-grey-color);
}

.modal-footer {
  padding: 15px 20px 20px;
  display: flex;
  justify-content: flex-end;
}

.modal-footer .modal-button {
  display: inline-block;
  box-sizing: border-box;
  text-align: center;
  border-radius: 4px;
  outline: none;
  cursor: pointer;
  min-width: 64px;
  height: 32px;
  padding: 0 16px;
  font-size: 14px;
  line-height: 30px;
  border-width: 1px;
  border-style: solid;
  background: none;
  color: var(--cancel-font-color);
  background-color: var(--cancel-background-color);
  border-color: var(--cancel-border-color);
  margin: 0 0 0 10px;
}

.modal-footer .modal-button.primary {
  color: var(--confirm-font-color);
  background-color: var(--confirm-background-color);
  border-color: var(--confirm-border-color);
}

.modal-footer .modal-button:hover {
  opacity: 0.8;
}

.modal-footer .modal-button:active {
  opacity: 0.6;
}

@media screen and (max-width: 540px) {
  .modal-box {
    width: calc(100% - 40px);
    margin: 0 20px;
  }

  .modal-header .modal-title {
    font-size: 18px;
  }

  .modal-content {
    font-size: 16px;
  }

  .modal-footer .modal-button {
    height: 38px;
    line-height: 38px;
    font-size: 16px;
  }
}
</style>
