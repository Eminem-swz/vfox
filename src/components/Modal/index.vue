<template>
  <div
    v-show="visible"
    :class="[prefix + '-modal']"
    :style="{ zIndex }"
    @click="onModalClick"
  >
    <div
      :class="[prefix + '-modal_box', { visible }]"
      :style="boxStyles"
      @click.stop="onBoxClick"
    >
      <div :class="[prefix + '-modal_header']" v-if="hasHeader">
        <slot name="header"></slot>
      </div>
      <div :class="[prefix + '-modal_body']">
        <slot></slot>
      </div>
      <div :class="[prefix + '-modal_footer']" v-if="hasFooter">
        <slot name="footer"></slot>
      </div>
      <i
        v-if="showClose"
        :class="[prefix + '-modal_close']"
        @click="onCloseClick"
        ><icon class-name="CloseOutlined"></icon
      ></i>
    </div>
  </div>
</template>

<script>
import Icon from '../Icon'
import { CustomEvent } from '../../helpers/events'
import { SDKKey } from '../../config'

export default {
  name: SDKKey + '-modal',
  components: { Icon },
  props: {
    width: {
      type: String,
      default: ''
    },
    visible: {
      type: Boolean,
      default: false
    },
    maskClosable: {
      type: Boolean,
      default: false
    },
    showClose: {
      type: Boolean,
      default: true
    },
    zIndex: {
      type: Number,
      default: 10000
    }
  },
  data() {
    return { prefix: SDKKey, hasHeader: false, hasFooter: false }
  },
  computed: {
    boxStyles() {
      const styles = {}

      if (this.width) {
        styles.width = this.width
      }

      return styles
    }
  },
  created() {},
  mounted() {
    if (this.$scopedSlots.header) {
      this.hasHeader = true
    }

    if (this.$scopedSlots.footer) {
      this.hasFooter = true
    }
  },
  beforeDestroy() {},
  methods: {
    onBoxClick() {},
    onModalClick(e) {
      if (this.maskClosable) {
        this.close(e)
      }
    },
    onCloseClick(e) {
      this.close(e)
    },
    close() {
      this.$emit('update:visible', false)

      const type = 'close'

      this.$emit(
        type,
        new CustomEvent(
          {
            type,
            currentTarget: this.$el
          },
          {}
        )
      )
    }
  }
}
</script>

<style lang="scss">
@import '../component.module.scss';

.#{$prefix}-modal {
  position: fixed;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  z-index: 10000;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: rgba($color: $black-color, $alpha: 0.5);
  text-align: left;
  transform: translateZ(0);

  &_box {
    width: 400px;
    box-sizing: border-box;
    border-radius: 4px;
    background-color: #fff;
    transform: scale(0);
    transition: all 0.2s;
    opacity: 0;

    &.visible {
      opacity: 1;
      transform: scale(1);
    }
  }

  &_header {
    flex: 1;
    font-size: 21px;
    font-weight: bold;
    color: $title-color;
    padding: 16px 0 12px;
    margin: 0 24px;
    display: -webkit-box;
    -webkit-line-clamp: 1;
    -webkit-box-orient: vertical;
    overflow: hidden;
    text-align: center;
  }

  &_body {
    padding: 0;
    font-size: 17px;
    line-height: 23.8px;
    color: $font-color;
  }

  &_footer {
    padding: 12px 0;
    display: flex;
    justify-content: space-between;
    margin: 0 24px;
  }

  &_close {
    position: absolute;
    left: 50%;
    bottom: -56px;
    margin: 0 0 0 -20px;
    background-color: rgba($color: $black-color, $alpha: 0.5);
    border: 2px solid #fff;
    width: 40px;
    height: 40px;
    border-radius: 50%;
    display: flex;
    align-items: center;
    justify-content: center;
    box-sizing: border-box;
    cursor: pointer;

    .#{$prefix}-icon {
      --size: 24px;
      --color: #fff;
    }
  }
}

@media screen and (max-width: 540px) {
  .#{$prefix}-modal {
    &_box {
      width: calc(100% - 40px);
      margin: 0 20px;
    }

    &_title {
      font-size: 18px;
    }

    &_content {
      font-size: 16px;
    }

    &_footer {
      .#{$prefix}-modal_button {
        height: 38px;
        line-height: 38px;
        font-size: 16px;
      }
    }
  }
}
</style>
