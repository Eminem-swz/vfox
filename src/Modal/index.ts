import { withInstall } from '@/helpers/with-install'
import Modal from './Modal.vue'

export { Modal }
export default withInstall(Modal)
