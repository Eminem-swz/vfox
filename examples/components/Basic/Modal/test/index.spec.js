/* global describe,test,expect */

import { mount } from '@vue/test-utils'
import Modal from '@/components/Modal'

describe('Modal', () => {
  test('should render default correctly', () => {
    const wrapper = mount(Modal, {})

    expect(wrapper.html()).toMatchSnapshot()
  })
})
