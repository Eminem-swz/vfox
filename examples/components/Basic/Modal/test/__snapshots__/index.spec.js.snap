// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render default correctly 1`] = `
<div class="fx-modal fx-popup" style="z-index: 2000; display: none;">
  <div class="fx-mask"></div>
  <div class="fx-modal_box">
    <div class="fx-modal_box-inner"></div> <i class="fx-modal_close"><svg aria-hidden="true" class="fx-icon">
        <use xlink:href="#icon-CloseCircleFilled"></use>
      </svg></i>
  </div>
</div>
`;
